include/mbedtls/error.h

```diff
@@ -127,6 +127,32 @@ extern "C" {
  */
 void mbedtls_strerror( int errnum, char *buffer, size_t buflen );
 
+/**
+ * \brief Translate high level part of a mbed TLS error code into a string
+ *        representation.
+ *
+ * This function returns a const pointer to an un-modifiable string. The caller
+ * must not try to modify the string and use it only for logging purposes.
+ *
+ * \param error_code    error code
```

Comment:
I wonder if instead of requiring a high-level error code, this function should accept any error code, and extract the high-level part.

Reply:
Done.

Reply:
Done.